optimum/habana/transformers/models/bart/modeling_bart.py

```diff
@@ -458,7 +458,9 @@ def gaudi_BartDecoder_forward(
 
     # past_key_values_length
     past_key_values_length = past_key_values[0][0].shape[2] if past_key_values is not None else 0
-    tensor_past_key_values_length = token_idx - 1 if use_cache else torch.tensor(past_key_values_length)
+    tensor_past_key_values_length = (
+        token_idx - 1 if (use_cache and token_idx) else torch.tensor(past_key_values_length)
```

Comment:
```suggestion
        token_idx - 1 if (use_cache and token_idx is not None) else torch.tensor(past_key_values_length)
```

Reply:
@malkomes Please test the bart with the changes you added here to make sure nothing new is introduced